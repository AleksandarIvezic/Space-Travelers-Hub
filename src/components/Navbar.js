import React from 'react';
import { NavLink } from 'react-router-dom';
import '../style/Navbar.css';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/rockets',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/my-profile',
      text: 'My profile',
    },

  ];
  return (
    <nav>
      <ul>
        {links.map((link) => (
          <li key={link.id}>
            <NavLink
              to={link.path}
              activeClassName="active-link"
              exact
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
