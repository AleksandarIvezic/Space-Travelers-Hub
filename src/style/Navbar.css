.line {
  content: '';
  height: 1.1rem;
  line-height: 1rem;
  border: 1px solid black;
}
