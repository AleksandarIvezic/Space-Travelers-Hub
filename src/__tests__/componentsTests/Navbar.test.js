import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../components/Navbar';

describe('Test if Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><Navbar /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
