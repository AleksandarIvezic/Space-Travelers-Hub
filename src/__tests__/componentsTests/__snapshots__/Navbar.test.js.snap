// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Navbar renders correctly 1`] = `
<nav
  className="d-flex flex-wrap justify-content-between align-items-center mx-4 mt-3"
>
  <div
    className="heading d-flex flex-wrap"
  >
    <img
      alt="logo"
      height="45"
      src="images/space-logo.png"
      width="45"
    />
    <h1
      className="mx-2"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    className="d-flex align-items-center list-unstyled m-0 p-0"
  >
    <li
      className="m-0 p-0 d-flex align-items-center"
    >
      <a
        aria-current="page"
        className="px-3 text-decoration-none active-link text-decoration-underline"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Rockets
      </a>
    </li>
    <li
      className="m-0 p-0 d-flex align-items-center"
    >
      <a
        aria-current={null}
        className="px-3 text-decoration-none"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <li
        className="line d-inline-block "
      />
    </li>
    <li
      className="m-0 p-0 d-flex align-items-center"
    >
      <a
        aria-current={null}
        className="px-3 text-decoration-none"
        href="/my-profile"
        onClick={[Function]}
      >
        My profile
      </a>
    </li>
  </ul>
</nav>
`;
